File: 61-rotate-list/rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (!head || !head->next || k == 0) return head;
        int n=1;
        auto temp=head;
        while(temp->next!=NULL){
            temp=temp->next;
            n++;
        }
        temp->next=head;
        k=k%n;
        int kl=0;temp=head;
        while(n-k-1>kl)
        {
            temp=temp->next;
            kl++;
        }
        head=temp->next;
        temp->next=NULL;
        return head;
    }
};

File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        
        if not head:
            return None
        lastElement=head
        length=1
        while lastElement.next:
            lastElement=lastElement.next
            length+=1
        lastElement.next=head

        k%=length
        tempNode=head
        for _ in range(length-k-1):
            tempNode=tempNode.next
        shiftk=tempNode.next
        tempNode.next=None

        return shiftk